### DISCUSSION TITLE
I have some trouble about file operation

### DISCUSSION THREAD

**Discussion by HUSTlwr** [exodrifter/unity-python#25] [Q&A] [answered]
I tried to creat a new .txt file and append content and there is my code:
`var engine = Python.CreateEngine();`
`var scope = engine.CreateScope();`
`string code = "";`
`code += "with open('Test1.txt', 'a') as test:" + "\n";`
`code += "\t" + "print('Test1', file = test)" + "\n";`
`var source = engine.CreateScriptSourceFromString(code);`
`source.Execute(scope);`

There is the exception infomation:
`SyntaxErrorException: unexpected token '='`
![z](https://user-images.githubusercontent.com/50614735/78658351-a5faf300-78fc-11ea-8a41-8177241bd8d8.png)

My unity version is 2018.3.11f1. I would appreciate if you can privide some clue. Thanks

**Comment by exodrifter** [comment, marked as answer]
The code you passed to `CreateScriptSourceFromString` is valid Python 3 syntax, but this plugin only supports Python 2 syntax.

The equivalent Python 2 syntax is:
```py
with open('Test1.txt', 'a') as test:
  test.write('Test1')
```

**Comment by HUSTlwr** [comment]
It's helpful. Thanks
